lib/sqlalchemy/sql/selectable.py

```diff
@@ -1549,7 +1564,7 @@ def _from_objects(self) -> List[FromClause]:
 
 
 class NoInit:
-    def __init__(self, *arg, **kw):
+    def __init__(self, *arg: Any, **kw: Any) -> NoReturn:
```

Comment:
this part, this is why all the _construct() calls are failing for mypy. 

Change this to:

```
def __init__(self, *arg: Any, **kw: Any):
```

that is, remove the "-> NoReturn" and leave it implicit.   Errors go from about 59 to 39 and are much simpler.  Can you make that change, then do a pass for some of the easier ones such as:

```
lib/sqlalchemy/sql/selectable.py:1654: error: Function is missing a return type annotation  [no-untyped-def]
lib/sqlalchemy/sql/selectable.py:1763: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
```

you can also add typing to the util functions that are outside:

```
lib/sqlalchemy/sql/selectable.py:4668: error: Call to untyped function "_expand_cloned" in typed context  [no-untyped-call]
lib/sqlalchemy/sql/selectable.py:4716: error: Call to untyped function "_cloned_intersection" in typed context  [no-untyped-call]
lib/sqlalchemy/sql/selectable.py:4717: error: Call to untyped function "_cloned_intersection" in typed context  [no-untyped-call]
lib/sqlalchemy/sql/selectable.py:4730: error: Call to untyped function "_cloned_difference" in typed context  [no-untyped-call]

```

Reply:
Thanks, I fixed NoReturn and lib/sqlalchemy/sql/selectable.py:1654,
lib/sqlalchemy/sql/selectable.py:1763
But I couldn't find a solution for the rest of mypy issues.
Perhaps we should accept the types we are sure about for now, and do the rest after the types have been added to other modules?

Reply:
yes perhaps!  just want to pare down what we have to a few errors :)    that is I want to make sure each "ignore" is really something that's not worth fixing.